packages/playwright/src/runner/reporters.ts

```diff
@@ -78,6 +77,13 @@ export async function createReporters(config: FullConfigInternal, mode: 'list' |
     else if (mode !== 'merge')
       reporters.unshift(!process.env.CI ? new LineReporter({ omitFailures: true }) : new DotReporter());
   }
+
+  if (!isTestServer && (mode === 'test' || mode === 'merge')) {
+    // If we are not in the test server, always add a last-run reporter.
+    const lastRunOutputFile = config.projects.length >= 0 ? path.join(config.projects[0].project.outputDir, '.last-run.json') : undefined;
+    reporters.push(new LastRunReporter({ ...runOptions, outputFile: lastRunOutputFile }));
```

Comment:
It kind of feels wrong that the `.last-run.json` is only written to the first project `outputDir`, but that's how it is at the moment.